tsl/src/nodes/vector_agg/vector_slot.h

```diff
@@ -36,7 +44,43 @@ static inline const CompressedColumnValues *
 vector_slot_get_compressed_column_values(TupleTableSlot *slot, const AttrNumber attnum)
 {
 	const uint16 offset = AttrNumberGetAttrOffset(attnum);
+
+	if (TTS_IS_ARROWTUPLE(slot))
+	{
+		ArrowTupleTableSlot *aslot = (ArrowTupleTableSlot *) slot;
+		const ArrowArray *arrow = arrow_slot_get_array(slot, attnum);
+		CompressedColumnValues *values = &aslot->ccvalues;
+
+		MemSet(values, 0, sizeof(CompressedColumnValues));
+
+		if (arrow == NULL)
+		{
+			bool isnull;
+
+			slot_getattr(slot, attnum, &isnull);
+			values->output_value = &slot->tts_values[offset];
+			values->output_isnull = &slot->tts_isnull[offset];
+			values->decompression_type = DT_Scalar;
+		}
+		else
+		{
+			values->arrow = (ArrowArray *) arrow;
+			values->buffers[0] = arrow->buffers[0];
+			values->buffers[1] = arrow->buffers[1];
+			values->decompression_type = TupleDescAttr(slot->tts_tupleDescriptor, offset)->attlen;
```

Comment:
The columnar scan has only by-value types now? Might be good to clarify in a comment and add an `Assert(values->decompression_type > 0)`.

Reply:
I guess I might not fully understand the need for or purpose of this decompression_type? In the current VectorAgg code, it only ever looks for DT_Scalar or a value > 0 (which is the fixed size of the data type). It seems to be something needed to decompress in lower nodes? But for aggregation we should only care about cases of arrow arrays.

Thefore, DT_Iterator should never happen because that means there is no arrow array.

 The values DT_ArrowText and DT_ArrowTextDict are never used or referenced in the VectorAgg code. I guess we can add that when needed?

Reply:
Ok, I'll add the missing types then.